src/dowel/csv_output.py

```diff
@@ -56,6 +54,36 @@ def record(self, data, prefix=''):
         else:
             raise ValueError('Unacceptable type.')
 
+    def _augment(self, data):
+        """Augment tabular data with new column(s)"""
+        data.reset()
+        self._log_file.close()
+
+        temp_file = self._file_name.split(".")
+        temp_file = "".join(temp_file[:-1]) + "_temp." + temp_file[-1]
+        os.rename(self._file_name, temp_file)
+        old_file = open(temp_file, 'r')
+        self._log_file = open(self._file_name, 'w')
+
+        self._fieldnames = data.as_primitive_dict.keys()
+        self._writer = csv.DictWriter(
+                self._log_file,
+                fieldnames=self._fieldnames,
+                extrasaction='ignore')
+
+        self._writer.writeheader()
+
+        reader = csv.DictReader(old_file)
+        old_fields = reader.fieldnames
+        augmented_elements = set(self._fieldnames).difference(set(old_fields))
+        for row in reader:
+            for k, v in row.items():
+                data.record(k, v)
+            self._writer.writerow(data.as_primitive_dict)
+            data.reset()
+        os.remove(temp_file)
```

Comment:
This works fine on Unix, but I think it will actually fail on Windows, since the file hasn't been closed yet.
Although Windows support isn't a priority by any means, it would be good to swap this with the next line.

Reply:
Are you referring to line 84? I can move the file closure on line 85 up.